onnxruntime/core/providers/webgpu/webgpu_execution_provider.cc

```diff
@@ -867,9 +859,11 @@ Status WebGpuExecutionProvider::OnRunEnd(bool /* sync_stream */, const onnxrunti
     context_.CollectProfilingData(profiler_->Events());
   }
 
-  if (IsPIXCaptureEnabled()) {
+#if defined(ENABLE_PIX_FOR_WEBGPU_EP)
+  if (context_.IsPixCaptureEnabled()) {
     context_.GeneratePIXFrame();
   }
+#endif  // ENABLE_PIX_FOR_WEBGPU_EP
```

Comment:
Suggest to hide implementation details in WebGpuExecutionProvider class:
- modify the name of `GeneratePIXFrame()` to `OnRunEnd()`, and always call it here.
- remove the `#if define` so that from the code it is an event handler calls another event handler, no detail is exposed.
- inside `WebGpuContext::OnRunEnd`, we can just rely on whether the unique_ptr is nullptr or not and no longer need to store 'enable_pix_capture'.